plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureStorageSettings.java

```diff
@@ -166,7 +184,6 @@ public LocationMode getLocationMode() {
     public String toString() {
         final StringBuilder sb = new StringBuilder("AzureStorageSettings{");
         sb.append("account='").append(account).append('\'');
-        sb.append(", key='").append(key).append('\'');
```

Comment:
add connectString instead here?

Reply:
@ywelsch this was actually an on purpose removal, do we really want to add secure credentials in `toString` methods?

Reply:
Unfortunately, for that we'd have to keep some field that holds this information. Maybe not worth it?

Reply:
+1